jdrfandroidbleparser/src/main/java/org/ehealthinnovation/jdrfandroidbleparser/bgm/characteristic/compoundcharacteristic/GlucoseMeasurementCharacteristic.kt

```diff
@@ -0,0 +1,142 @@
+     * The year, month, day, hour, minute, second have value according to https://www.bluetooth.com/specifications/gatt/viewer?attributeXmlFile=org.bluetooth.characteristic.date_time.xml
+     * These are the raw value extracted from the binary packet and not directly usable without proper conversion.
+     * These values are converted into a [Date] as [displayTime]
+     */
+    var timeOffset: Int? = null
+    var timeYear: Int? = null
+    var timeMonth: Int? = null
+    var timeDay: Int? = null
+    var timeHour: Int? = null
+    var timeMinute: Int? = null
+    var timeSecond: Int? = null
+    var displayTime: Date? = null
+
+    /**
+     * The concentration is the raw value from the binary packet. It has unit stored in [unit].
+     * A convenient function [getConcentration] to get the value in desired unit can be used to query this value.
+     */
+    var concentration: Float? = null
+    var unit: Units? = null
+
+    /**
+     * Contains a set of flag concerning the annunciation. Before interpreting the glucose concentration, make sure
+     * no flags indicating measurement error is present in this EnumSet.
+     */
+    var annunciationFlags: EnumSet<SensorStatusAnnunciation>? = null
+
+    /**
+     * The location and type of the blood sample.
+     */
+    var fluidType: Type? = null
+    var sampleLocation: SampleLocation? = null
+
+
+    /**
+     * This function is called when creating an characteristic to parse the binary array stream
+     */
+    override fun parse(c: BluetoothGattCharacteristic): Boolean {
+        var errorFreeParse = false
+        try {
+            //make sure c is not null
+            Flags.parsFlags(getNextIntValue(c, FormatType.FORMAT_UINT8.formatType)).let {
+                flags = it
+                sequenceNumber = getNextIntValue(c, FormatType.FORMAT_UINT16.formatType)
+                timeYear = getNextIntValue(c, FormatType.FORMAT_UINT16.formatType)
+                timeMonth = getNextIntValue(c, FormatType.FORMAT_UINT8.formatType)
+                timeDay = getNextIntValue(c, FormatType.FORMAT_UINT8.formatType)
+                timeHour = getNextIntValue(c, FormatType.FORMAT_UINT8.formatType)
+                timeMinute = getNextIntValue(c, FormatType.FORMAT_UINT8.formatType)
+                timeSecond = getNextIntValue(c, FormatType.FORMAT_UINT8.formatType)
+
+                if (it.contains(Flags.TIME_OFFSET_PRESENT))
+                    timeOffset = getNextIntValue(c, FormatType.FORMAT_UINT16.formatType)
+                if (it.contains(Flags.CONCENTRATION_PRESENT)) {
+                    concentration = getNextFloatValue(c, FormatType.FORMAT_SFLOAT.formatType)
+                    val tempIntHolder = getNextIntValue(c, FormatType.FORMAT_UINT8.formatType)
+                    fluidType = Type.fromKey(tempIntHolder and 0x0F)
+                    sampleLocation = SampleLocation.fromKey((tempIntHolder and 0xF0) shr 4)
+                    if (it.contains(Flags.CONCENTRATION_UNIT))
```

Comment:
I know it is most likely never going to happen, but Glucose concentration units are a separate flag from the concentration, so pull the if condition out and populate it separately. It will help with testing and readability.

Also add braces.